invenio_stats/aggregations.py

```diff
@@ -263,52 +266,67 @@ def _get_oldest_event_timestamp(self):
             return None
         return parser.parse(result[0]["timestamp"])
 
-    def agg_iter(self, lower_limit, upper_limit):
+    def _split_date_range(self, lower_limit, upper_limit):
+        res = {}
+        current_interval = lower_limit
+        delta = INTERVAL_DELTAS[self.interval]
+        while current_interval < upper_limit:
+            dt_key = current_interval.strftime(SUPPORTED_INTERVALS[self.interval])
+            res[dt_key] = current_interval
+            current_interval += delta
+
+        dt_key = upper_limit.strftime(SUPPORTED_INTERVALS[self.interval])
+        res[dt_key] = upper_limit
+        return res
+
+    def agg_iter(self, dt):
         """Aggregate and return dictionary to be indexed in ES."""
-        aggregation_data = {}
-        start_date = format_range_dt(lower_limit, self.interval)
-        end_date = format_range_dt(upper_limit, self.interval)
+        rounded_dt = format_range_dt(dt, self.interval)
         agg_query = dsl.Search(using=self.client, index=self.event_index).filter(
-            "range", timestamp={"gte": start_date, "lte": end_date}
+            "range", timestamp={"gte": rounded_dt, "lte": rounded_dt}
```

Comment:
self note: up and lower limits are the same so then it returns a day or a month or a (one unit) of the interval.